tests/integration/whl_filegroup/BUILD.bazel

```diff
@@ -0,0 +1,21 @@
+load("@rules_cc//cc:defs.bzl", "cc_library", "cc_test")
+load("@rules_python//python:pip.bzl", "whl_filegroup")
+
+whl_filegroup(
+    name = "numpy_includes",
+    pattern = "numpy/core/include/numpy",
+    whl = "@pypi//numpy:whl",
+)
```

Comment:
Right now this test won't be executed as part of CI and I would love to ensure that this does not break.

We could create a `whl` in the build context, similarly to how the `examples/wheel` is doing and then use the `whl_filegroup` to do the extracting. We could still include some headers in the `whl` file and have a `cc_library` and `cc_test` as a verification and document that this is mimicking a usecase where `numpy` headers need to be used.

The main recommendation here would be to avoid testing in a separate workspace. :)

It could be also nice to add some documentation in `docs/sphinx/`.

Reply:
Yea, CI also fails due to stumbling over these files and not seeing @pypi// somehow. Need to look into this, not sure what's different from the other integration tests?

Reply:
Moved the test next to the implementation into the same workspace

Reply:
Thanks!

FYI, we usually keep the bzl tests in the tests directory, that way we can have more clear distinction about what is production and non production code.

At least that is true for all new code.

Reply:
Yes, please move all the test related parts under //tests somewhere, probably a tests/whl_filegroup directory.

Part of the "production vs non production" code distinction is to avoid adding dependencies on test-only code in production code paths. e.g. a BUILD file that loads() something only for testing incurs having to download those repos, parse those packages, parse those loaded bzl files, etc.

Reply:
done